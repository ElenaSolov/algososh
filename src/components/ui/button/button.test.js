import { Button } from "./button";
import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent, screen } from "@testing-library/react";

describe("Button renders correctly", () => {
  it("Button without text renders correctly", () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("Button with text renders correctly", () => {
    const button = renderer.create(<Button text="Test" />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("Disabled Button is disabled", () => {
    const button = renderer
      .create(<Button text="Test" disabled={true} />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it("Loader renders correctly and button with loader is disabled", () => {
    const button = renderer
      .create(<Button text="Test" isLoader={true} />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it("Button click fires callback", () => {
    const onClick = jest.fn();
    render(<Button text="Test" onClick={onClick} />);
    const button = screen.getByText("Test");
    fireEvent.click(button);
    expect(onClick).toHaveBeenCalled();
  });
});
