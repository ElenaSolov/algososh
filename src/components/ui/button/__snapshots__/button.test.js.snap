// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendered correctly Button with text rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Test
  </p>
</button>
`;

exports[`Button rendered correctly Button without text rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button rendered correctly Disabled Button is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Test
  </p>
</button>
`;

exports[`Button rendered correctly Loader rendered correctly and button with loader is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
